File: python/pca/testpca.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# vim:fenc=utf-8
#
# Copyright © 2020 Julian Bustamante <jbustamante35@gmail.com>
#
# Distributed under terms of the MIT license.

"""
Use my Pcajb class
"""
from magic import *
from Pcajb import Pcajb
import matplotlib.pyplot as plt

N   = 99
nx  = 3
#sel = [0,2,4]
sel = np.arange(0,N)
X   = np.array(magic(N))
Z   = np.append(X, X[0,:]).reshape(N+1, N)
px  = Pcajb(Z,nx)

x   = px.Data
#m,u = px.meanSubtract()
#c   = px.covarMatrix()
#e,v = px.eigens()
#s   = px.PCAScores(sel)
#V,T = px.VarExplained(0.8)
xx  = px.SimData()

#print(np.shape(xx))
#print(xx)
#print(np.shape(x))
#print(x)
#print(V)
#print(T)
#plt.scatter(x[:,1], xx[:,1])
fig,axs = plt.subplots(1,2, constrained_layout=True)
axs[0].imshow(x)
ttl = 'Input Data\n[%d N | %d mag | %d nx]' % (np.shape(Z)[0],N,nx)
axs[0].set_title(ttl)
axs[1].imshow(xx)
ttl = "Simulated Data\n[%d PCs]" % (nx)
axs[1].set_title(ttl)
plt.show()

File: r/pca/testpca.R
# -*- coding: utf-8 -*-
# vim:fenc=utf-8
## ---------------------------
## testpca.R
##
## Author: Julian Bustamante
##
## Date Created: 2020-03-05
##
## Copyright (c) Julian Bustamante, 2020
## Email: <jbustamante35@gmail.com>
##
## Distributed under terms of the MIT license.
##
## ---------------------------
##
## Notes:
##      Use my custom PCA class
##
## ---------------------------

setwd("/home/jbustamante/Dropbox/ComputerProgramming/programminglanguages/r/pca")
library('pracma')
source('Pcajb.R')

N  = 99
n  = 3
x  = magic(N)
px = Pcajb(Data=x, npc=n)

#mu = px$meanSubtract()
#m  = Reshape(unlist(mu[1]), size(x,1), size(x,2))
#u  = Reshape(unlist(mu[2]), 1, size(x,2))
#
#cv = px$covarMatrix()
#
#wv = px$eigens(n)
#w  = Reshape(unlist(wv[1]), 1, n)
#v  = Reshape(unlist(wv[2]), N, n)
#
#s = px$PCAScores()
z = px$SimData()

plot.new()
image(x)
#image(z)



